src/util-device.c

```diff
@@ -109,21 +110,96 @@ int LiveRegisterDeviceName(const char *dev)
         return -1;
     }
 
+    if (copy_dev) {
+        pd->copy_dev = SCStrdup(copy_dev);
+        if (unlikely(pd->copy_dev == NULL)) {
+            SCFree(pd->dev);
+            SCFree(pd);
+            return -1;
+        }
+    }
+
+    pd->role = SCStrdup(ROLE_UNKNOWN_STR);
+    if (unlikely(pd->role == NULL)) {
+        SCFree(pd->dev);
+        SCFree(pd);
+        return -1;
+    }
+
+    TAILQ_INSERT_TAIL(&pre_live_devices, pd, next);
+
+    SCLogDebug("Device \"%s\" with role \"%s\" registered.", dev, ROLE_UNKNOWN_STR);
+    return 0;
+}
+
+/**
+ *  \brief Add a device for monitoring
+ *
+ * To be used during option parsing. When a device has
+ * to be created during runmode init, use LiveRegisterDevice()
+ * Differs from LiveRegisterDeviceName by allowing for configurable
+ * role of a given device. Limited to devices/interfaces defined in
+ * suricata.yaml.
+ *
+ *  \param dev string with the device name
+ *  \param copy_dev string with the name of the device being copied to (valid in IPS mode only)
+ *  \param role string with the device role
+ *
+ *  \retval 0 on success.
+ *  \retval -1 on failure.
+ */
+int LiveRegisterDeviceNameAndRole(const char *dev, const char *copy_dev, const char *role)
+{
+    LiveDeviceName *pd = NULL;
+
+    pd = SCCalloc(1, sizeof(LiveDeviceName));
+    if (unlikely(pd == NULL)) {
+        return -1;
+    }
+
+    pd->dev = SCStrdup(dev);
+    if (unlikely(pd->dev == NULL)) {
+        SCFree(pd);
+        return -1;
+    }
+
+    if (copy_dev) {
+        pd->copy_dev = SCStrdup(copy_dev);
+        if (unlikely(pd->copy_dev == NULL)) {
+            SCFree(pd->dev);
+            SCFree(pd);
+            return -1;
+        }
+    }
+
+    if (unlikely(role == NULL)) {
+        pd->role = SCStrdup(ROLE_UNKNOWN_STR);
+    } else {
+        pd->role = SCStrdup(role);
+    }
+
+    if (unlikely(pd->role == NULL)) {
+        SCFree(pd->dev);
+        SCFree(pd);
+        return -1;
+    }
+
     TAILQ_INSERT_TAIL(&pre_live_devices, pd, next);
 
-    SCLogDebug("Device \"%s\" registered.", dev);
+    SCLogDebug("Device \"%s\" with role \"%s\" registered.", dev, role);
     return 0;
 }
 
 /**
  *  \brief Add a pcap device for monitoring and create structure
  *
  *  \param dev string with the device name
+ *  \param role string with the device role
  *
  *  \retval 0 on success.
  *  \retval -1 on failure.
  */
-int LiveRegisterDevice(const char *dev)
+int LiveRegisterDevice(const char *dev, const char *role)
```

Comment:
Similarly, suggest leaving this interface alone and adding `LiveRegisterDeviceWithRole`.